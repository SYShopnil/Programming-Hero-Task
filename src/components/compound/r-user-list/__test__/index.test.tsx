import { render, screen } from "@testing-library/react";
import { RUserList } from "..";
import { EDataTestId } from "@src/types/common";

describe("When UserList has rendered", () => {
  it("Expect render successfully", async () => {
    const { getByTestId } = render(await RUserList()); // ARRANGE
    const myElement = getByTestId(EDataTestId.rUserList);
    // const myElement = screen.getByTestId(EDataTestId.rUserList);
    expect(myElement).toMatchSnapshot();
  });
});
