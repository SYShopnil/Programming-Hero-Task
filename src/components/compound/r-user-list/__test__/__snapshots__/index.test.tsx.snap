// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When UserList has rendered Expect render successfully 1`] = `
<div
  class="flex justify-center items-center"
  data-testid="rUserList"
>
  <div
    class="w-full lg:w-1/2 bg-gray-100 rounded-lg shadow-md"
  >
    <div
      class="bg-gray-700 text-white py-2 px-4"
    >
      <div
        class="flex"
      >
        <div
          class="w-3/6 text-center"
        >
          Email
        </div>
        <div
          class="w-1/6 text-center"
        >
          User Name
        </div>
        <div
          class="w-1/6 text-center"
        >
          Profile Picture
        </div>
        <div
          class="w-1/6 text-center"
        >
          User Type
        </div>
        <div
          class="w-1/6 text-center"
        >
          Gender
        </div>
      </div>
    </div>
    <div
      class="py-2 px-4"
    >
      <div
        data-testid="RTableBody"
      >
        <div
          class="flex bg-white border-b border-gray-300"
        >
          <div
            class="w-3/6 text-center"
          >
            sadmanishopnil@gmail.com
          </div>
          <div
            class="w-1/6 text-center"
          >
            John Doe
          </div>
          <div
            class="w-1/6 text-center"
          >
            <img
              alt="Profile Pic"
              class="w-10 h-10 mx-auto rounded-full"
              src="/assert/default-profile.png"
            />
          </div>
          <div
            class="w-1/6 text-center"
          >
            user
          </div>
          <div
            class="w-1/6 text-center"
          >
            male
          </div>
        </div>
        <div
          class="flex bg-gray-100 border-b border-gray-300"
        >
          <div
            class="w-3/6 text-center"
          >
            sadmanishopnil1@gmail.com
          </div>
          <div
            class="w-1/6 text-center"
          >
            Jane Smith
          </div>
          <div
            class="w-1/6 text-center"
          >
            <img
              alt="Profile Pic"
              class="w-10 h-10 mx-auto rounded-full"
              src="/assert/default-profile.png"
            />
          </div>
          <div
            class="w-1/6 text-center"
          >
            admin
          </div>
          <div
            class="w-1/6 text-center"
          >
            male
          </div>
        </div>
        <div
          class="flex bg-white border-b border-gray-300"
        >
          <div
            class="w-3/6 text-center"
          >
            xyz@gmail.com
          </div>
          <div
            class="w-1/6 text-center"
          >
            Harry Rose
          </div>
          <div
            class="w-1/6 text-center"
          >
            <img
              alt="Profile Pic"
              class="w-10 h-10 mx-auto rounded-full"
              src="/assert/default-profile.png"
            />
          </div>
          <div
            class="w-1/6 text-center"
          >
            user
          </div>
          <div
            class="w-1/6 text-center"
          >
            female
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
